web/modules/custom/dpl_event/src/Plugin/rest/resource/v1/EventResource.php

```diff
@@ -114,6 +115,19 @@ public function patch(string $uuid, Request $request): Response {
     }
 
     $state = $request_data->getState();
+
+    if ($state) {
+      $state_enum = EventState::tryFrom($state);
+
+      if (is_null($state_enum)) {
+        $this->logger->error("Could not map '@value' to EventState enum.", [
+          '@value' => $state,
+        ]);
+      }
+
+      $state = $state_enum?->value;
+    }
+
```

Comment:
Please reconsider this change.

As I see we must notify the client and reject the request if an invalid value is passed. Otherwise: How are they going to know that they have done something wrong? We could also log the error internally but as I see it this is less important.

```suggestion
    if ($state) {
      try {
        $state_enum = EventState::from($state);
      } catch (\ValueError $e) {
        throw new \Symfony\Component\HttpFoundation\Exception\BadRequestException("Invalid value for state: {$state}")
      }
    }
```